### PR TITLE
Migrated test_dir_gfid_heal_on_all_subvols.py

### PR DESCRIPTION
**Pull request by aujjwal-redhat**
Migrated
[test_dir_gfid_heal_on_all_subvols.py](https://github.com/gluster/glusto-tests/blob/HEAD/tests/functional/afr/test_dir_gfid_heal_on_all_subvols.py)

Updates: #292

Signed-off-by: Ayush Ujjwal <aujjwal@redhat.com>

### CODE REVIEW

**Comment by aujjwal-redhat** [comment]
```sh
✔ test_heal_command_unsuccessful_as_bricks_down-rep Succeeded
Results for afr
Test nature : disruptive
TestName : test_heal_command_unsuccessful_as_bricks_down
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     rep     |     PASS    |        00:10:30       |      NA     |
+-------------+-------------+-----------------------+-------------+
```

**Comment by nik-redhat** [comment]
Is it draft or ready, @aujjwal-redhat ?

**Comment by aujjwal-redhat** [comment]
> Is it draft or ready, @aujjwal-redhat ?

This is still draft. The other one is ready.. I mistakenly added this as ready : |

File: common/ops/support_ops/io_ops.py [hunk 1]
```diff
@@ -563,6 +564,7 @@ def get_mounts_stat(self, mounts: list) -> bool:
                 f"Stat of mount {mount['client']}:{mount['mountpath']}")
             cmd = f"python3 /tmp/file_dir_ops.py stat -R {mount['mountpath']}"
             ret = self.execute_abstract_op_node(cmd, mount['client'], False)
+            sleep(10)
```
Comment (srijan-sivakumar) [inline common/ops/support_ops/io_ops.py:567]: What's the purpose behind the sleep ?

Comment (nik-redhat) [reply common/ops/support_ops/io_ops.py:567]: To let the lookup, complete restoring the gfid..

Comment (nik-redhat) [reply common/ops/support_ops/io_ops.py:567]: It sometimes, work with lesser time as well, but sometimes it does fail.

**Comment by nik-redhat** [comment]
```
Results for afr
Test nature : disruptive
TestName : test_dir_gfid_heal_on_all_subvols
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     rep     |     PASS    |        00:00:31       |      NA     |
|   dist-rep  |     PASS    |        00:00:36       |      NA     |
|     dist    |     FAIL    |        00:00:29       |      NA     |
+-------------+-------------+-----------------------+-------------+
+--------------------------------------+-------------------+
|               Category               |       Count       |
+--------------------------------------+-------------------+
|           Disruptive Tests           |         1         |
|         Non Disruptive Tests         |         0         |
|       Disruptive Tests Skipped       |         0         |
|     Non Disruptive Tests Skipped     |         0         |
|              Tests Ran               |         1         |
|            Tests Skipped             |         0         |
|             Total Tests              |         1         |
|   Disruptive Tests Pass Percentage   | 66.66666666666667 |
| Non Disruptive Tests Pass Percentage |         0         |
|        Total Pass Percentage         | 66.66666666666667 |
+--------------------------------------+-------------------+
Total Time Taken : 00 hours 01 minutes 47 seconds
✔ Tearing down successful.
```

File: common/ops/support_ops/io_ops.py (hunk 1 above)
Comment (nik-redhat) [reply common/ops/support_ops/io_ops.py:567]: Oh, sorry...this sleep is not needed. I didn't check this file while updating the PR.

Comment (srijan-sivakumar) [reply common/ops/support_ops/io_ops.py]: cool

**Comment by nik-redhat** [comment]
We are not able to find a fix issue like, when we remove the gfid of the first brick then the TC  passes for all types of volume (including dist), but fails otherwise.

**Comment by srijan-sivakumar** [comment]
> We are not able to find a fix issue like, when we remove the gfid of the first brick then the TC passes for all types of volume (including dist), but fails otherwise.

Do you mean to say that "when the gfid removal fails for on vol type, it fails for all" ?

**Comment by nik-redhat** [comment]
> > We are not able to find a fix issue like, when we remove the gfid of the first brick then the TC passes for all types of volume (including dist), but fails otherwise.
> 
> Do you mean to say that "when the gfid removal fails for on vol type, it fails for all" ?

No, as per the TC code, it passes for all of them except `dist` volume type. But, if we remove the gfid oof the first brick instead of the second, thrid brick the TC passes for dist as well.

**Comment by srijan-sivakumar** [comment]
> > > We are not able to find a fix issue like, when we remove the gfid of the first brick then the TC passes for all types of volume (including dist), but fails otherwise.
> > 
> > 
> > Do you mean to say that "when the gfid removal fails for on vol type, it fails for all" ?
> 
> No, as per the TC code, it passes for all of them except `dist` volume type. But, if we remove the gfid oof the first brick instead of the second, thrid brick the TC passes for dist as well.

🤔 intresting..

**Comment by nik-redhat** [comment]
I guess we can take this TC in, and let it fail for `dist` volume. Logically, everything seems to be fine.

**Comment by nik-redhat** [comment]
I have tested in the glusto env, as well. The TC fails for dist vol only. I think we are good to merge the TC.

**Review by Sheetalpamecha — APPROVED** [review]

**Pull request merged**